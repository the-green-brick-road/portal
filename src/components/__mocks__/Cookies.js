/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
