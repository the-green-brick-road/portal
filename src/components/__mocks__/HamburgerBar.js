/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
