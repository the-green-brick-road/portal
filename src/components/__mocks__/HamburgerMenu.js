/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;