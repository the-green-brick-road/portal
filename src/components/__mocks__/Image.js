/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
