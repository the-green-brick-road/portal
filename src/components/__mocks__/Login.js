/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
