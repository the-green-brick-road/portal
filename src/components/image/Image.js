/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
