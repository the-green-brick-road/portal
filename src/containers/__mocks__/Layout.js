/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment }   from 'react';
import { Outlet }     from "react-router-dom";

/* Material UI includes */
import { Typography } from '@mui/material';

function Layout(props) {

    return (

        <Fragment>
            <Typography> Mock Layout </Typography>
            <Outlet />
        </Fragment>

    );

}

export default Layout;
