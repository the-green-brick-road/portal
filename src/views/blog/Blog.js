/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Blog page definition
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 31 may 2023
# ---------------------------------------------------- */

/* React includes */
import { useEffect, Fragment, useState }  from 'react';

/* Material UI includes */
import { Link, Card, CardContent, CardMedia, CardHeader, Container, Typography, Paper } from '@mui/material';
import { Grid } from '@mui/material';
import { useTheme }                                                     from '@mui/material/styles';

/* Portal includes */
import { useData, useDesign }                                           from '../../providers';

function Blog() {

    /* --------- Gather inputs --------- */
    const { posts }                     = useData();
    const { sizes }                     = useDesign();
    const theme                         = useTheme();
    const [ localPosts, setLocalPosts ] = useState([])

    const months   = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
    // const componentName = 'Blog'

    useEffect(() => {

        /* ------- Sort posts by date ------ */
        const local_posts = [];
        for (let i_post = 0; i_post < posts.length; i_post +=1) {

            if (local_posts.length === 0) { local_posts.push(posts[i_post]); }
            else {

                let j_post = 0
                if(posts[i_post].date.seconds > local_posts[local_posts.length - 1].date.seconds) { local_posts.push(posts[i_post]); }
                else {

                    while((posts[i_post].date.seconds > local_posts[j_post].date.seconds) && ((j_post + 1) < local_posts.length)) { j_post += 1 }
                    local_posts.splice(j_post, 0, posts[i_post]);

                }

            }

        }
        setLocalPosts(local_posts)

    }, [posts]);

    /* ----------- Define HTML --------- */
    return (
        <Fragment>
            <Container style={{ backgroundColor: theme.palette.common.black, width:'100%', height:sizes['menu-height']}}/>
            <Container style={{ backgroundColor: theme.palette.common.black, width:'100%', overflowX: 'hidden', overflowY: 'hidden' }}>
                <Grid container spacing={1} columns={60} style={{ transformOrigin: '50vw 50vh' }} >

                    { localPosts.map((item, index) => { /* Loop on all menu content to create button */

                        let url=""
                        let target = ""
                        if( item.real ) {

                            if ('url' in item ) { url = item.url; target="_blank"}
                            else { url = `/posts/${item.id}` }

                        }

                        var time = new Date(1970, 0, 1); // Epoch
                        time.setSeconds(item.date.seconds);
                        const month = months[time.getMonth()];
                        const year = time.getYear() + 1900;
                        const date = time.getDate();

                        return(
                            <Grid item xs={30} sm={20} md={15} lg={12} key={index} style={{padding:0}}>

                                {(item.real) && (
                                    <Link href={url} underline="none" target={target}>
                                        <Card elevation={4} style={{ backgroundColor:theme.palette.common.white, margin:10 }}>
                                            <CardMedia component="img" style={{objectFit: 'cover', objectPosition: item['image-position'] }} image={item.image}/>
                                            <CardContent style={{ backgroundColor:theme.palette.common.white, padding:'5px' }}>
                                                <Paper style={{ backgroundColor:theme.palette.primary.main, paddingLeft:'5px', paddingRight:'5px' }} >
                                                    <Typography variant="body1" style={{ color:theme.palette.common.white, fontSize:'11px', fontWeight:'bold'}}> {date}-{month}-{year} </Typography>
                                                </Paper>
                                            </CardContent>
                                            <CardHeader style={{ backgroundColor:theme.palette.common.white, padding:'5px'}} titleTypographyProps={{ fontSize:'12px', fontWeight:'bold', fontFamily:item.font }} title={item.title} />
                                        </Card>
                                    </Link>
                                )}
                                {(!item.real) && (
                                    <Card elevation={4} style={{ backgroundColor:theme.palette.common.white, margin:10 }}>
                                        <CardMedia component="img" style={{objectFit: 'cover', objectPosition: item['image-position'] }} image={item.image}/>
                                        <CardContent style={{ backgroundColor:theme.palette.common.white, padding:'5px' }}>
                                            <Paper style={{ backgroundColor:'#888888', paddingLeft:'5px', paddingRight:'5px' }} >
                                                <Typography variant="body1" style={{ color:theme.palette.common.white, fontSize:'11px', fontWeight:'bold'}}> {date}-{month}-{year} </Typography>
                                            </Paper>
                                        </CardContent>
                                        <CardHeader style={{ backgroundColor:theme.palette.common.white, padding:'5px'}} titleTypographyProps={{ fontSize:'12px', fontWeight:'bold', fontFamily:item.font }} title={item.title} />
                                    </Card>
                                )}
                            </Grid>
                        )

                    })}
                </Grid>
            </Container>
        </Fragment>
    );

}

export default Blog;