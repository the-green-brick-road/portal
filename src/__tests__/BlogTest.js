/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Home view component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM }       from '@testing-library/react'
import { expect, test}             from '@jest/globals';

/* Component under test */
import { default as Blog }         from '../views/blog/Blog';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { default as MockImage }    from '../components/__mocks__/Image';
jest.mock("../components", () => ({ Image: (props) => MockImage(props) }));
/* eslint-enable jest/no-mocks-import */

describe("Blog view" ,() => {

    test('Should display blog page', async () => {

        const view = render(
            <div>
                <Blog/>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

})
