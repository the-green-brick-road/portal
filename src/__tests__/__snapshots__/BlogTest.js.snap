// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog view Should display blog page 1`] = `
"<body>
  <div>
    <div>
      <p
        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
      >
        Blog
      </p>
    </div>
  </div>
</body>"
`;
